client/src/fluid/FluidAutocomplete.ml

```diff
@@ -328,12 +328,32 @@ let secretToACItem (s : SecretTypes.t) : fluidAutocompleteItem =
   let asDval = DStr (Util.obscureString s.secretValue) in
   FACVariable (s.secretName, Some asDval)
 
+let isInQuery (tl : toplevel) ti =
+  let ast' = TL.getAST tl in
+  match ast' with
+  | None -> false
+  | Some ast ->
+    FluidAST.ancestors (FluidToken.tid ti.token) ast
+    |> List.find ~f:(function FluidExpression.EFnCall (_, name, _, _) -> if name = "DB::query_v4" then true else false | _ -> false)
+    |> Option.is_some
+
+let filterToDbSupportedFns (tl : toplevel) ti functions =
+  if not (isInQuery tl ti) then
+    functions
+  else
+    functions
+    |> List.filter ~f:(fun f -> 
+      match f with
+      | FACFunction fn -> fn.fnIsSupportedInQuery
+      | _ -> false)
```

Comment:
I think you need to allow other things, such as pipes and variables (I don't know the full list, the SQL compiler source will probably know).

Reply:
pipes and variables are working because they are included in separate lets (keywords and varnames respectively I think) and concat'd together in generateExprs.  In this context I'm pretty sure we will only ever see FACFunctions.